VariantEncoding.md

```diff
@@ -399,6 +399,7 @@ The Decimal type contains a scale, but no precision. The implied precision of a
 | Timestamp            | timestamp with time zone   | `22`    | TIMESTAMP(isAdjustedToUTC=true, NANOS)       | 8-byte little-endian                                                                        |
 | TimestampNTZ         | timestamp without time zone | `23`    | TIMESTAMP(isAdjustedToUTC=false, NANOS)      | 8-byte little-endian                                                                        |
 | UUID                 | uuid                        | `24`    | UUID                         | 16-byte big-endian                                                                         |
+| Fixed(L)             | Byte array of length L | `25`    | FIXED_LEN_BYTE_ARRAY[L]     |  4 byte little-endian size L, followed by length-L big-endian bytes    |
```

Comment:
Why using `big-endian` bytes?

Reply:
Why do we need the size? Shouldn't that be in the type description?

Reply:
big-endian bytes: this is to keep in sync with the others like UUID which is a fixed(16). And it makes sense to write a  bytes in big endian since the engine can write the bytes in the buffer in order, not requiring buffering the whole string.

The required size: I initially avoided adding the fixed(L) type because I believed we couldn't support fixed(L) if we try to include L in the type description, as there wouldn't be enough bits available to represent the length, given that only 5 bits are allocated for the type.

The way here to add the fixed(L) type is to add the length in the value field - we are duplicating the length for each value but I don't see other ways.

Reply:
- I'm not sure endianness makes sense for fixed(L), endianess only applies to multi-bytes structures?  Fixed(L) each bytes is independent.

- I think the current proposal is reasonable and matches how things like decimal with arbitrary precisions are encoded.  It is also consistent with string representation, if we are worried about overhead of 4 bytes then we could use a variable width encoding schema (or have two types Short-fixed(L) with 1 byte and fixed(L) with 4 buytes.  Unfortunately, IIUC we can't have a 'short-fixed L' like we have for string because I think we are already use the entire number range there.

Reply:
Re: Size - makes sense, i forgot we are storing type information for every row anyway so it probably makes more sense to store it here in the value section of the Variant.

I agree with @emkornfield that we probably don't need to specify endianess.

Reply:
Yeah. Make sense that we don't need endianness.

Reply:
@emkornfield number range - you mean the type range? We can have from 0 - 31 with a few left. I think it makes sense to to add Short-fixed(L). Let me know if you agree.

> we can't have a 'short-fixed L' like we have for string because I think we are already use the entire number range there.

Reply:
> @emkornfield number range - you mean the type range? We can have from 0 - 31 with a few left. I think it makes sense to to add Short-fixed(L). Let me know if you agree.
> 
> > we can't have a 'short-fixed L' like we have for string because I think we are already use the entire number range there.

I was talking about the range for "basic type", I'm OK adding it as a separate "primitive type" though.

Reply:
Oh. I see. Yeah. We have used all of them for "basic type". We can't add like short-string. 

Let me make the change and share with you.